datafusion/common/src/dfschema.rs

```diff
@@ -314,6 +329,26 @@ impl DFSchema {
         }
     }
 
+    /// Find if the field exists with the given name
+    pub fn has_column_with_unqualified_name(&self, name: &str) -> bool {
+        let matches = self.fields_with_unqualified_name(name);
```

Comment:
Since we are doing an optimization exercise, I think we should avoid the `collect()` into a Vec as well (as that also allocates)

Perhaps something like:

```suggestion
  self.fields
            .iter()
            .any(|field| {
                field.qualifier().map(|q| q.eq(qualifier)).unwrap_or(false)
                    && field.name() == name
            })
```
(maybe you can factor out the check into a function to avoid the duplication)